### ISSUE TITLE
"filename" missing for attachments that have `path` only

### ISSUE THREAD

**Issue by niftylettuce** [nodemailer/nodemailer#799]
Using the `jsonTransport: true` to test attachments, it appears that `filename` is not automatically attached to the response `message` object.  e.g. if you specify `attachments: [ { path: '/path/to/some-actual-file.txt' } ]` it will have the following object after `sendMail` is called.

```js
  "attachments": [
    {
      "content": "cm9vdCA9Iaowiejfaoiwjefoiajwefoijawefiojawoefijawoeifj",
      "encoding": "base64"
    }
  ],
```

Note that `filename` is missing, but we advertise in the docs here <https://nodemailer.com/message/attachments/> that its automatically added.

> // filename and content type is derived from path

**Comment by andris9** [comment]
Could you try if the [latest commit](5d77f26a865b1fe15026102d8cc1db3ac4ee0e6c) does what you'd expect from the JSON transport output?

**Comment by niftylettuce** [comment]
works great @andris9 

let me know when you release to npm?

also check out <https://github.com/niftylettuce/preview-email>

perhaps we can put it under official "nodemailer" org and mention it in docs?

**Comment by andris9** [comment]
Published as v4.1.1.

Initiate a repository transfer to move the repo to the Nodemailer org. I can then accept it and add your user as a collaborator for that repo.

**Comment by niftylettuce** [comment]
thanks for doing this, I released `preview-email@0.0.3`, here's a demo which has inline attachment links as well you can test/open.

![demo](https://user-images.githubusercontent.com/523141/31860336-6cc9ccb2-b6e6-11e7-930b-88cd89959c02.png)
